pkg/monitoring/rules/recordingrules/operator.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2024 The KubeVirt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recordingrules
+
+import (
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	"github.com/machadovilaca/operator-observability/pkg/operatorrules"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	abnormalLabelKey = "reason"
+)
+
+var operatorRecordingRules = []operatorrules.RecordingRule{
+	{
+		MetricsOpts: operatormetrics.MetricOpts{
+			Name: "kubevirt_working_set_memory_exceeded",
+			Help: "The pod with the highest exceeded memory for each container based on the working set.",
+			ConstLabels: map[string]string{
+				abnormalLabelKey: "WorkingSetMemoryExceeded",
+			},
+		},
+		MetricType: operatormetrics.GaugeType,
+		Expr:       intstr.FromString("max by(container, namespace)(container_memory_working_set_bytes{container=~\"virt-controller|virt-api|virt-handler|virt-operator\"}  - on(pod) group_left(node) (kube_pod_container_resource_requests{container=~\"virt-controller|virt-api|virt-handler|virt-operator\",resource=\"memory\"}))"),
+	},
+	{
+		MetricsOpts: operatormetrics.MetricOpts{
+			Name: "kubevirt_rss_memory_exceeded",
+			Help: "The pod with the highest exceeded memory for each container based on the rss.",
+			ConstLabels: map[string]string{
+				abnormalLabelKey: "RSSMemoryExceeded",
+			},
+		},
+		MetricType: operatormetrics.GaugeType,
+		Expr:       intstr.FromString("max by(container, namespace)(container_memory_rss{container=~\"virt-controller|virt-api|virt-handler|virt-operator\"}  - on(pod) group_left(node) (kube_pod_container_resource_requests{container=~\"virt-controller|virt-api|virt-handler|virt-operator\",resource=\"memory\"}))"),
+	},
+	{
+		MetricsOpts: operatormetrics.MetricOpts{
+			Name: "cnv_abnormal",
```

Comment:
this should be `kubevirt_abnormal`, and even then what do you think about adding something else to it:
`kubevirt_abnormal_state`, for example, IMO would read better